libs/datasets/timeseries.py

```diff
@@ -540,10 +541,24 @@ def from_csv(path_or_buf: Union[pathlib.Path, TextIO]) -> "MultiRegionDataset":
     def read_from_pointer(pointer: dataset_pointer.DatasetPointer) -> "MultiRegionDataset":
         wide_dates_df = pd.read_csv(pointer.path_wide_dates(), low_memory=False)
         wide_dates_df = wide_dates_df.set_index([CommonFields.LOCATION_ID, PdFields.VARIABLE])
+
         # Extract provenance column from wide date DataFrame so all columns are dates.
-        # TODO(tom): support multiple provenance columns
-        provenance_series = wide_dates_df[PdFields.PROVENANCE].dropna()
-        wide_dates_df = wide_dates_df.drop(columns=[PdFields.PROVENANCE])
+        provenance_column_mask = wide_dates_df.columns.str.match(
```

Comment:
Can you add a comment about why this is a match?

Reply:
Done